File: app/src/main/java/com/robillo/readrush/data/db/model/search/SearchName.java
package com.robillo.readrush.data.db.model.search;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

/**
 * Created by robinkamboj on 29/11/17.
 */

@Entity(tableName = "searchname")
public class SearchName {

    @SuppressWarnings("WeakerAccess")
    @NonNull
    @PrimaryKey
    String mSearchName;

    public SearchName(@NonNull String mSearchName) {
        this.mSearchName = mSearchName;
    }

    @NonNull
    public String getmSearchName() {
        return mSearchName;
    }

    @SuppressWarnings("unused")
    public void setmSearchName(@NonNull String mSearchName) {
        this.mSearchName = mSearchName;
    }
}

File: app/src/main/java/com/robillo/readrush/data/db/model/search/SearchName.kt
package com.robillo.readrush.data.db.model.search

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

/**
 * Created by robinkamboj on 29/11/17.
 */

@Entity
class SearchName(@field:PrimaryKey
                 var mSearchName: String) {

    fun getmSearchName(): String {
        return mSearchName
    }

    fun setmSearchName(mSearchName: String) {
        this.mSearchName = mSearchName
    }
}
